addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -351,6 +352,41 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                             }
                             break;
                         }
+                        case WAKEUP: {
+                            if (command instanceof OnOffType) {
+                                if (((OnOffType) command) == OnOffType.ON) {
+                                    wakeUp();
+                                }
+                            }
+                            break;
+                        }
+                        case ALLOWWAKEUP: {
+                            if (command instanceof OnOffType) {
+                                if (((OnOffType) command) == OnOffType.ON) {
+                                    allowWakeUp = true;
+                                } else {
+                                    allowWakeUp = false;
+                                }
+                            }
+                            break;
+                        }
+                        case ENABLEEVENTS: {
+                            if (command instanceof OnOffType) {
```

Comment:
It looks that this event thread already existed before, but does that mean that still starts unconditionally? Should the condition apply there as well? Or can you only change it after the binding is up-and-running?

Reply:
Yes, the idea is that is starts anyways when the Thing is initialized, and that it can only be stopped afterwards

Reply:
Can you also extend the readme to explain this behaviour, to me it's also still not really clear so is there no permanent way to disable it or would you need rules / persistence for that. Isn't this were we have configuration for?

Cc: @kaikreuzer

Reply:
@martinvw I refer to the discussion in one of the logged Issues. The desire for other users of the binding is to be able to steer the behaviour of the Thing from the Rules. AFAIK that can only be done through Channels, since Configuration is static (from the point of view of Rules).

This Thread is started at initialisation of the ThingHandler, and after startup, it can be started/stopped through ENABLEEVENTS. Now, since at initialisation of the Handler, no Command has been fed to the Handler, it can not know whether the Thread has to be started or not (since that is not part of Configuration, but dynamically fed by the UI or Rules or...), and therefore the choice is to start the Thread because this event stream is an essential part of the communication between the binding and the Tesla back-end servers. Some users want to control this event stream because of concerns over power usage by the vehicle (e.g. switch it off to conserve battery power)

Reply:
I agree that the better place would be the Thing configuration.

> since Configuration is static (from the point of view of Rules).

It is indeed not possible so far to change a configuration through rules (I'd also consider something like that an "admin operation"). But I can also live with adding it as a channel for now, if we consider it as a "feature" of the car that it let's the user toggle whether it sends events or not.

But please also update the README.md, thanks!